addons/account_edi_ubl_cii/tests/test_ubl_cii.py

```diff
@@ -219,3 +221,27 @@ def test_import_partner_peppol_fields(self):
         # The partner should be retrieved based on the peppol fields
         imported_invoice = self.import_attachment(xml_attachment, self.company_data["default_journal_sale"])
         self.assertEqual(imported_invoice.partner_id, partner)
+
+    def test_export_pdf_contains_facturx(self):
+        """ Check that we generate a Factur-x xml when we render the invoice action report. """
+        invoice = self.env['account.move'].create({
+            'partner_id': self.partner_a.id,
+            'move_type': 'out_invoice',
+            'invoice_line_ids': [Command.create({'product_id': self.product_a.id})]
+        })
+        invoice.action_post()
+        observer = []
+
+        def _export_invoice(*args, **kwargs):
+            observer.append('We generated a Factur-X xml !')
+            return b'<xml>Tmp</xml>', {}
+
+        with patch(
+            'odoo.addons.account_edi_ubl_cii.models.account_edi_xml_cii_facturx.AccountEdiXmlCII._export_invoice',
+            side_effect=_export_invoice,
+        ):
+            self.env['ir.actions.report']\
+                .with_context(force_report_rendering=True)\
+                ._render('account.account_invoices', invoice.ids)
+
+        self.assertEqual(observer, ['We generated a Factur-X xml !'])
```

Comment:
I think you can simplify this with `assert_called_once()` on the patched method.